src/lib/components/objects/GLTF.svelte

```diff
@@ -0,0 +1,65 @@
+<script>
+	import { onMount } from 'svelte';
+
+	import * as THREE from 'three';
+	import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader.js';
+	import { DRACOLoader } from 'three/examples/jsm/loaders/DRACOLoader.js';
+    import { KTX2Loader } from 'three/examples/jsm/loaders/KTX2Loader';
+	import { setup } from '../../utils/context.js';
+	import { transform } from '../../utils/object.js';
+	import * as defaults from '../../utils/defaults.js';
+
+	/** @type {string} */
+	export let path = null;
+	/** @type {string} */
+	export let filename = null;
+	export let useDraco = false;
+	export let useKTX2 = false;
+
+	export let position = defaults.position;
+	export let rotation = defaults.rotation;
+	export let scale = defaults.scale;
+	export let castShadow = false;
+	export let receiveShadow = false;
+	export let frustumCulled = true;
+	export let renderOrder = 0;
+
+	/** @type {Object} */
+	let gltfScene = null;
+
+	const { root, parent, self } = setup(new THREE.Object3D());
+
+	$: {
+		if(gltfScene) {
```

Comment:
Could/should this instead be executed on successful load from GLTFLoader?
Maybe worth dispatching on load as well.

Reply:
Can you give an example of what that might look like?

Reply:
Sure! 
At the top of the file before/after the props are exported, we could create an event dispatcher:
```js
import { createEventDispatcher } from "svelte";
const dispatch = createEventDispatcher();
```
And then on [line 77](https://github.com/Rich-Harris/svelte-cubed/pull/25/files#diff-881ff166de0eb6fb0745293f466a247ab3aa642fe5dab6f27960f794ebee8285R77) where we have successfully loaded (or failed to load) the model, we could `dispatch("load", gltf);` or `dispatch("error", err);`, etc...

My thinking was that this could provide a convenient entry point for manipulating the materials in the scene, etc...

Reply:
As a consumer of the component, that would look something like
```svelte
<SC.GLTF {...} on:load={swapMyTexture} on:error={swapFallbackScene} />
```

Reply:
I'm not sure that dispatching on change would provide any value to _me_, but but the implementation would be similar, just within the reactive statement.